airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/source.py

```diff
@@ -23,38 +23,136 @@
+                )
+                selector = {
+                    "fields": ["Name", "CanManageClients", "CustomerId", "TestAccount", "DateTimeZone", "CurrencyCode"],
+                    "predicates": [
+                        {
+                            "field": "CustomerId",
+                            "operator": "IN",
+                            "values": [
+                                customer_id,
+                            ],
+                        }
+                    ],
+                }
+                accounts = self._get_accounts(logger, sdk_client, selector)
+                if accounts:
+                    account = accounts[0]
+                    is_manager = account.canManageClients
+                    for stream in streams:
+                        if stream.endswith("REPORT") and is_manager:
+                            logger.log_by_prefix(f"Unable to sync {stream} with the manager account {customer_id}", "ERROR")
+                            sys.exit(1)
+                else:
+                    logger.log_by_prefix(f"Unable to sync with the provided customer id {customer_id}", "ERROR")
+                    sys.exit(1)
+        except Exception as err:
+            logger.log_by_prefix(f"Unable to sync. Error: {err}", "ERROR")
+            sys.exit(1)
+
     def check_config(self, logger: AirbyteLogger, config_path: str, config: json) -> AirbyteConnectionStatus:
+        # singer catalog that attempts to pull a stream ("accounts") that should always exists, though it may be empty.
         try:
-            # singer catalog that attempts to pull a stream ("accounts") that should always exists, though it may be empty.
-            singer_check_catalog_path = os.path.abspath(os.path.dirname(__file__)) + "/singer_check_catalog.json"
-            read_cmd = self.read_cmd(logger, config_path, singer_check_catalog_path)
-            if SingerHelper.read(logger, read_cmd) is not None:
-                return AirbyteConnectionStatus(status=Status.SUCCEEDED)
-            else:
-                return AirbyteConnectionStatus(status=Status.FAILED)
+            customer_ids = config["customer_ids"].split(",")
+            for customer_id in customer_ids:
+                oauth2_client = oauth2.GoogleRefreshTokenClient(
+                    config["oauth_client_id"], config["oauth_client_secret"], config["refresh_token"]
+                )
 
+                sdk_client = adwords.AdWordsClient(
+                    config["developer_token"], oauth2_client, user_agent=config["user_agent"], client_customer_id=customer_id
+                )
+                selector = {
+                    "fields": ["Name", "CanManageClients", "CustomerId", "TestAccount", "DateTimeZone", "CurrencyCode"],
+                }
+                accounts = self._get_accounts(logger, sdk_client, selector)
+                if not accounts:
+                    return AirbyteConnectionStatus(status=Status.FAILED)
```

Comment:
We should always have a message set if the connection check fails. What should the error message here say?